docs/flows/flows.puml

```diff
@@ -272,20 +271,136 @@ bitdaoadmin <-- insurancecontract: execute proposal
 
 newpage
 
-autonumber "Upgrades:[00]"
+autonumber "UpgradeGovernance:[00]"
 
-== Upgrades ==
+== Upgrade Governance ==
 
-bitdaoadmin -> governanceproxy: deploy proxy and set initial implementation
-bitdaoadmin -> treasuryproxy: deploy proxy and set initial implementation
-bitdaoadmin -> executorproxy: deploy proxy and set initial implementation
+bitdaoadmin -> executor: deploy timelock executor
+bitdaoadmin -> bitdaogovernance: deploy initial governance contract logic
+bitdaoadmin -> proxyadmin: deploy admin for governance proxy
+bitdaoadmin -> proxy: deploy governance proxy and set logic and admin
+bitdaoadmin -> proxyadmin: set executor as admin
 
 
 alice -> bitdaogovernance: deploy new bitdaogovernance implementation
+
+alice -> forum: create forum upgrade proposal with new bitdaogovernance
+alice <- community: positive/negative feedback
+
+community -> community: delegate role (developer/all) based votes
+
+alice -> bitdaogovernance: submit upgrade proposal
+community -> bitdaogovernance: roles voting
+
+bitdaoadmin -> bitdaogovernance: get roles voting result
+bitdaoadmin <-- bitdaogovernance: pass/fail vote result
+
+bitdaoadmin -> bitdaogovernance: start execution
+executor <-- bitdaogovernance: execute proposal
+executor -> proxyadmin: call governance proxy upgrade to new bitdaogovernance implementation
+
+newpage
+
+autonumber "UpgradeTreasury:[00]"
+
+== Upgrade Treasury ==
+
+bitdaoadmin -> bitdaotreasury: deploy initial treasury contract logic, set governance
+bitdaoadmin -> proxyadmin: deploy admin for treasury proxy
+bitdaoadmin -> proxy: deploy treasury proxy and set logic and admin
+bitdaoadmin -> proxyadmin: set governance executor as admin
+
+
 alice -> bitdaotreasury: deploy new bitdaotreasury implementation
-alice -> executor: deploy new executor implementation
 
-alice -> forum: create forum upgrade proposal
+alice -> forum: create forum upgrade proposal with new bitdaogovernance
+alice <- community: positive/negative feedback
+
+community -> community: delegate role (treasury/developer/all) based votes
+
+alice -> bitdaogovernance: submit upgrade proposal
+community -> bitdaogovernance: roles voting
+
+bitdaoadmin -> bitdaogovernance: get roles voting result
+bitdaoadmin <-- bitdaogovernance: pass/fail vote result
+
+bitdaoadmin -> bitdaogovernance: start execution
+executor <-- bitdaogovernance: execute proposal
+executor -> proxyadmin: call treasury proxy upgrade to new bitdaotreasury implementation
+
+newpage
+
+autonumber "UpgradeVault:[00]"
+
+== Upgrade Vault ==
+
+bitdaoadmin -> vault: deploy initial vault contract logic, set treasury
+bitdaoadmin -> proxyadmin: deploy admin for vault proxy
+bitdaoadmin -> proxy: deploy vault proxy and set logic and admin
+bitdaoadmin -> proxyadmin: set governance executor as admin
+
+
+alice -> vault: deploy new vault implementation
+
+alice -> forum: create forum upgrade proposal with new bitdaogovernance
+alice <- community: positive/negative feedback
```

Comment:
nit: double minus sign for a dotted return message line

UML message return lines are usually dotted rather than solid.